rad/src/operators/string.rs

```diff
@@ -35,6 +38,113 @@ pub fn parse_json_array(input: &RadonString) -> Result<RadonArray, RadError> {
     item.try_into()
 }
 
+fn add_children(
+    map: &mut BTreeMap<String, RadonTypes>,
+    text_children: Vec<RadonTypes>,
+    element_children: Vec<(String, RadonTypes)>,
+) {
+    match text_children.len() {
+        0 => {}
+        1 => {
+            let text_value = text_children.into_iter().next().unwrap();
+            map.insert("_text".to_string(), text_value);
+        }
+        _ => {
+            let text_value = RadonArray::from(text_children).into();
+            map.insert("_text".to_string(), text_value);
+        }
+    }
+
+    for (key, value) in element_children {
+        match map.get_mut(&key) {
+            Some(old_value) => match old_value {
+                RadonTypes::Array(rad_array) => {
+                    let mut new_array = rad_array.value();
+                    new_array.push(value);
+
+                    *rad_array = RadonArray::from(new_array);
+                }
+                x => {
+                    let new_array = vec![x.clone(), value];
+                    *x = RadonArray::from(new_array).into();
+                }
+            },
+            None => {
+                map.insert(key, value);
+            }
+        }
+    }
+}
+
+fn parse_element_map(
+    input: &minidom::Element,
+    use_namespace: bool,
+    depth: u8,
+) -> Result<RadonTypes, RadError> {
+    if depth > MAX_DEPTH {
+        return Err(RadError::XmlParseOverflow);
+    }
+
+    let mut map: BTreeMap<String, RadonTypes> = BTreeMap::new();
+    if use_namespace {
+        map.insert("_xmlns".to_string(), RadonString::from(input.ns()).into());
+    }
+    for (k, v) in input.attrs() {
+        map.insert(format!("@{}", k), RadonString::from(v).into());
+    }
+
+    let mut element_children = vec![];
+    let mut text_children: Vec<RadonTypes> = vec![];
+    for child in input.nodes() {
+        match child {
+            minidom::Node::Element(elem) => {
+                let key = elem.name().to_string();
+                let value = parse_element_map(elem, false, depth + 1)?;
+
+                element_children.push((key, value));
+            }
+            minidom::Node::Text(text) => {
+                // This check is to avoid blank spaces in xml would be included in the RadonMap
+                let text_var = text.trim().to_string();
+                if !text_var.is_empty() {
+                    text_children.push(RadonString::from(text_var).into());
+                }
+            }
+        }
+    }
+
+    let only_text_children = element_children.is_empty() && map.is_empty();
+
+    if only_text_children && text_children.is_empty() {
+        let text_value = RadonString::from("").into();
+        Ok(text_value)
+    } else if only_text_children {
+        let text_value = text_children.into_iter().next().unwrap();
```

Comment:
This branch has an implicit assertion that when `only_text_children` is true, there will only be 1 element in `text_children` vector. How about we make it explicit? Or handle that case somehow.

```suggestion
        assert_eq!(text_children.len(), 1);
        let text_value = text_children.into_iter().next().unwrap();
```